src/test/unit/EigenPodManagerUnit.t.sol

```diff
@@ -117,220 +85,438 @@ contract EigenPodManagerUnitTests is Test, EigenPodPausingConstants {
+
+        // Add shares for user
+        cheats.startPrank(address(delegationManagerMock));
+        eigenPodManager.addShares(podOwner, shares);
+
+        // Remove shares
+        eigenPodManager.removeShares(podOwner, shares);
+        cheats.stopPrank();
+
+        // Check storage update
+        assertEq(eigenPodManager.podOwnerShares(podOwner), 0);
+    }
+
+    /*******************************************************************************
+                        WithdrawSharesAsTokens Tests
+    ******************************************************************************/
+
+    function test_withdrawSharesAsTokens_revert_podOwnerZeroAddress() public {
+        cheats.prank(address(delegationManagerMock));
+        cheats.expectRevert("EigenPodManager.withdrawSharesAsTokens: podOwner cannot be zero address");
+        eigenPodManager.withdrawSharesAsTokens(address(0), address(0), 0);
+    }
+
+    function test_withdrawSharesAsTokens_revert_destinationZeroAddress() public {
+        cheats.prank(address(delegationManagerMock));
+        cheats.expectRevert("EigenPodManager.withdrawSharesAsTokens: destination cannot be zero address");
+        eigenPodManager.withdrawSharesAsTokens(defaultStaker, address(0), 0);  
+    }
+
+    function testFuzz_withdrawSharesAsTokens_revert_sharesNegative(int256 shares) public {
+        cheats.assume(shares < 0);
+        cheats.prank(address(delegationManagerMock));
+        cheats.expectRevert("EigenPodManager.withdrawSharesAsTokens: shares cannot be negative");
+        eigenPodManager.withdrawSharesAsTokens(defaultStaker, defaultStaker, uint256(shares));
+    }
+
+    function testFuzz_withdrawSharesAsTokens_revert_sharesNotWholeGwei(uint256 shares) public {
+        cheats.assume(int256(shares) >= 0);
+        cheats.assume(shares % GWEI_TO_WEI != 0);
+
+        cheats.prank(address(delegationManagerMock));
+        cheats.expectRevert("EigenPodManager.withdrawSharesAsTokens: shares must be a whole Gwei amount");
+        eigenPodManager.withdrawSharesAsTokens(defaultStaker, defaultStaker, shares);
+    }
+
+    /**
+     * @notice The `withdrawSharesAsTokens` is called in the `completeQueuedWithdrawal` function from the 
+     *         delegationManager. When a withdrawal is queued in the delegationManager, `removeShares is called`
+     */
+    function test_withdrawSharesAsTokens_reduceEntireDeficit() public {
+        // Shares to initialize & withdraw
+        int256 sharesBeginning = -100e18;
+        uint256 sharesToWithdraw = 101e18;
+        
+        // Deploy Pod And initialize with negative shares
+        _initializePodWithShares(defaultStaker, sharesBeginning);
+
+        // Withdraw shares
+        cheats.prank(address(delegationManagerMock));
+        eigenPodManager.withdrawSharesAsTokens(defaultStaker, defaultStaker, sharesToWithdraw);
+
+        // Check storage update
+        assertEq(eigenPodManager.podOwnerShares(defaultStaker), int256(0));
+    }
```

Comment:
it would be cool if after this (and similarly in the `test_withdrawSharesAsTokens_withdrawPositive` test) we could check that the contract is correctly calling `EigenPod.withdrawRestakedBeaconChainETH` ... might be a more general problem to sort out a solution to, definitely not going to block this PR on this

Reply:
Planning to include that in the integration test

Reply:
> Planning to include that in the integration test

makes sense. it would kind of be stretching what a unit test should cover.